examples/with-supabase/app/_examples/route-handler-static/route.ts

```diff
@@ -0,0 +1,23 @@
+// TODO: Duplicate or move this file outside the `_examples` folder to make it a route
+
+import { createClient } from '@supabase/supabase-js'
+import { NextResponse } from 'next/server'
+
+export async function GET() {
+  // This uses `supabase-js` directly as the request happens
+  // at build-time, rather than when the user requests the route,
+  // therefore, there are no cookies to pass through with the request
+  // to Supabase.
+  const supabase = createClient(
```

Comment:
IMO it's more confusing to show both ways – any reason to not pick the first as the canonical way?

Reply:
Using the Supabase Auth Helpers for Next.js package, you need to import the `cookies` function, which makes the route dynamic, and currently fails if you don't mark the route as dynamic with something like `export const dynamic = 'force-dynamic'`

To use Supabase in a static route, you need to use `supabase-js`

Reply:
Using the Supabase Auth Helpers for Next.js package, you need to import the cookies function, which makes the route dynamic, and currently fails the build if you don't mark the route as dynamic with something like export const dynamic = 'force-dynamic'

To use Supabase in a static route, you need to use supabase-js

Reply:
Using the Supabase Auth Helpers for Next.js package, you need to import the `cookies` function, which makes the route dynamic, and currently fails the build if you don't mark the route as dynamic with something like `export const dynamic = 'force-dynamic'`

To use Supabase in a static route, you need to use `supabase-js`

Reply:
Good point about an ever-growing examples folder becoming confusing though. I have removed all examples, except for creating a Supabase client in each of the Next.js App Router contexts:

- Client Component
- Server Component
- Route Handler
- Server Action